spi_nand_flash/src/spi_nand_oper.c

```diff
@@ -26,7 +27,17 @@ esp_err_t spi_nand_execute_transaction(spi_device_handle_t device, spi_nand_tran
         .dummy_bits = transaction->dummy_bits
     };
 
-    return spi_device_transmit(device, (spi_transaction_t *) &e);
+    if (transaction->flags == SPI_TRANS_USE_TXDATA && transaction->mosi_len <= 4) {
```

Comment:
I would recommend something like

```c
if (transaction->flags == SPI_TRANS_USE_TXDATA) {
    assert(transaction->mosi_len <= 4 && "SPI_TRANS_USE_TXDATA used for a long transaction")
```

The point is, this is a sanity check for the way we call `spi_nand_execute_transaction` from other source files. If we accidentally pass length > 4 along with SPI_TRANS_USE_TXDATA, we want to notice this immediately. But the way it is right now, the part inside `if` will be skipped and we will just not pass any data to SPI transaction.